packages/camera/camera_android_camerax/android/src/test/java/io/flutter/plugins/camerax/CameraStateTest.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import androidx.camera.core.CameraState;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.CameraStateFlutterApi;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.CameraStateType;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.CameraStateTypeData;
+import java.util.Objects;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class CameraStateTest {
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Mock public CameraState mockCameraState;
+  @Mock public BinaryMessenger mockBinaryMessenger;
+  @Mock public CameraStateFlutterApi mockFlutterApi;
+
+  InstanceManager instanceManager;
+
+  @Before
+  public void setUp() {
+    instanceManager = InstanceManager.create(identifier -> {});
+  }
+
+  @After
+  public void tearDown() {
+    instanceManager.clear();
+  }
+
+  @Test
+  public void flutterApiCreate_makesCallToDartToCreateInstance() {
+    final CameraStateFlutterApiWrapper flutterApi =
+        new CameraStateFlutterApiWrapper(mockBinaryMessenger, instanceManager);
+    flutterApi.setApi(mockFlutterApi);
+
+    final CameraStateType type = CameraStateType.OPEN;
+    final CameraState.StateError mockError = mock(CameraState.StateError.class);
+
+    flutterApi.create(mockCameraState, type, mockError, reply -> {});
+
+    final long instanceIdentifier =
+        Objects.requireNonNull(instanceManager.getIdentifierForStrongReference(mockCameraState));
+    final ArgumentCaptor<CameraStateTypeData> cameraStateTypeDataCaptor =
+        ArgumentCaptor.forClass(CameraStateTypeData.class);
+
+    verify(mockFlutterApi)
+        .create(
+            eq(instanceIdentifier),
+            cameraStateTypeDataCaptor.capture(),
+            eq(Objects.requireNonNull(instanceManager.getIdentifierForStrongReference(mockError))),
+            any());
+
+    assertEquals(cameraStateTypeDataCaptor.getValue().getValue(), type);
+  }
+
+  @Test
+  public void getCameraStateType_returnsExpectedType() {
+    for (CameraState.Type type : CameraState.Type.values()) {
```

Comment:
No action needed: 
Another patten you might see in code would reduce the duplicate calls to  `assertEquals(CameraStateFlutterApiWrapper.getCameraStateType(type), CameraStateType.<THING>);`. It can be easier to read if you break the conditions into a map (or pair) of {CameraState.Type to expected CameraStateType} that validates the known conversions. Then another test that loops over CameraState.Type.values() and verifies that all the values are non null and do not throw an exception.

Reply:
Added to https://github.com/flutter/flutter/issues/125928